fuel-core/src/config/chain_config.rs

```diff
@@ -121,6 +120,16 @@ pub struct StateConfig {
     pub height: Option<BlockHeight>,
 }
 
+impl StateConfig {
+    pub fn generate_state_config(db: Database) -> Self {
+        StateConfig {
+            coins: db.get_coin_config().unwrap_or_default(),
+            contracts: db.get_contract_config().unwrap_or_default(),
+            height: db.get_block_height().unwrap_or_default(),
```

Comment:
This method should propagate the errors from the database methods without unwrap. Otherwise, the snapshot could silently fail while still providing partially valid results without the user knowing.

```suggestion
    pub fn generate_state_config(db: Database) -> anyhow::Result<Self> {
        StateConfig {
            coins: db.get_coin_config()?,
            contracts: db.get_contract_config()?,
            height: db.get_block_height()?,
```

Reply:
I initially did this so corruption in one column would allow the rest to be saved, but I see how this silent failure could be more harmful

Reply:
This could be enabled in the future using a `continue-on-error` flag, however, any errors should be reported via `error!` or `warn!` before `unwrap_or_default` so the user is aware that their snapshot may come with some issues.